lib/rules/selector-no-type/README.md

```diff
@@ -50,7 +50,25 @@ The following patterns are *not* considered warnings:
 
 ## Optional secondary options
 
-### `ignore: ["compounded", "descendant"]`
+### `ignore: ["child", "compounded", "descendant"]`
+
+#### `"child"`
+
+Allow child selectors.
+
+The following patterns are *not* considered warnings:
+
+```css
+a > foo {}
+```
+
+```css
+a > #bar {}
+```
+
+```css
+#bar > a {}
+```
```

Comment:
Whoops, I totally missed that.

> Either I'm not understanding the intention, or something is wrong here. 

You are correct.

> only addressed type selectors following child combinators 

I think this is the correct behaviour and I think is consistent with how the current `"descendant"` option works. Although, it'd be nice to have some tests [here](https://github.com/stylelint/stylelint/blob/2ddb4a5bf3ef5d7518d599bddeb5a210797d9323/lib/rules/selector-no-type/__tests__/index.js#L114-L120) to definitely show that things like `a .foo {}` and `a div {}` are rejected.